install_config/router/default_haproxy_router.adoc

```diff
@@ -1457,6 +1457,129 @@ access any pod in the cluster. If isolation is needed in this case, then do not
+configuration.
+
+The dynamic configuration internally uses the haproxy socket and configuration
+API with a pool of pre-allocated routes and back end servers. The pre-allocated
+pool of routes is created using route blueprints. The default set of blueprints
+supports unsecured routes, edge secured routes (without any custom TLS
+configuration), and passthrough routes.
+
+[IMPORTANT]
+====
+`re-encypt` routes require custom TLS configuration information and so are not
+supported out of the box.
+
+However, the blueprints that the dynamic configuration manager uses can be
+extended by setting the `ROUTER_BLUEPRINT_ROUTE_NAMESPACE` and optionally the
+`ROUTER_BLUEPRINT_ROUTE_LABELS` environment variable values. All routes (or the
+routes matching the route labels) in the blueprint route namespace will be
+processed as custom blueprints similar to the default set of blueprints. This
+enables defining certain types of routes such as `re-encypt` routes or routes
+that use custom annotations or routes with custom TLS configuration as
+blueprints.
+====
+
+The following procedure assumes you create three routes: `reencrypt-blueprint`, `annotated-edge-blueprint`, and `annotated-unsecured-blueprint`.
+
+.Procedure
+
+. Create a new project:
++
+----
+$ oc new-project namespace_name
+----
+
+. Create a new route:
++
+----
+$ oc create route edge edge_route_name --key=/path/to/key.pem \
+      --cert=/path/to/cert.pem --service=<service> --port=8443
+----
+
+. Label the route:
++
+----
+$ oc label route edge_route_name type=route_label_1
+----
+
+. xref:../../dev_guide/routes.adoc#creating-routes[Create three different routes]. All have the label `type=route_label_1`:
++
+----
+$ oc create -f reencrypt-blueprint.yaml
+$ oc create -f annotated-edge-blueprint.yaml
+$ oc create -f annotated-unsecured-blueprint.json
+----
++
+You can also remove a label from a route, which prevents it from being used as a
+blueprint route. For example, to prevent the `annotated-unsecured-blueprint`
+from being used as a blueprint route:
++
+----
+$ oc label route annotated-unsecured-blueprint type-
+----
+
+. Create a new router:
+----
+$ oc adm router
+----
+
+. Set the environment variables for the new router:
++
+----
+$ oc set env dc/router ROUTER_HAPROXY_CONFIG_MANAGER=true      \
+                       ROUTER_BLUEPRINT_ROUTE_NAMESPACE=namespace_name  \
+                       ROUTER_BLUEPRINT_ROUTE_LABELS="type=route_label_1"
+----
+
+All routes in the namespace or project `namespace_name` with label
+`type=route_label_1` are able to be processed and used as custom blueprints.
+
+Note that you can also add, update, or remove blueprints by managing the routes
+as you would normally in that namespace `namespace_name`. The dynamic
+configuration manager watches for changes to routes in the namespace
+`namespace_name` similar to how the router watches for `routes` and `services`.
+
+The pool sizes of the pre-allocated routes and back end servers can be
```

Comment:
This should be a separate step. Like, "Optionally, set blueprint management parameters..." then a ul of the parameters and sample command.